test/gha/cloudformation/iam_cloudformation.yaml

```diff
@@ -0,0 +1,475 @@
+              - ssm:UpdateInstanceInformation
+            Resource: "*"
+          - Effect: Allow
+            Action:
+              # SSM Permissions for AmazonSSMManagedInstanceCore policy applied to the NodeInstanceRole
+              - ssmmessages:CreateControlChannel
+              - ssmmessages:CreateDataChannel
+              - ssmmessages:OpenControlChannel
+              - ssmmessages:OpenDataChannel
+            Resource: "*"
+          - Effect: Allow
+            Action:
+              # SSM Permissions for AmazonSSMManagedInstanceCore policy applied to the NodeInstanceRole
+              - ec2messages:AcknowledgeMessage
+              - ec2messages:DeleteMessage
+              - ec2messages:FailMessage
+              - ec2messages:GetEndpoint
+              - ec2messages:GetMessages
+              - ec2messages:SendReply
+            Resource: "*"
+          - Effect: Allow
+            Action:
+              - sqs:DeleteMessage
+              - sqs:GetQueueAttributes
+              - sqs:GetQueueUrl
+              - sqs:SendMessage
+              - sqs:ReceiveMessage
+            Resource: "*"
+          - Effect: Allow
+            Action: iam:PassRole
+            Resource:
+              - !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/KarpenterNodeRole-*"
+              - !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/aws-fis-itn"
+          - Effect: Allow
+            Action:
+             - pricing:GetProducts
+             - ec2:DescribeSpotPriceHistory
+            Resource: "*"
+          - Effect: Allow
+            Action: eks:DescribeCluster
+            Resource: "*"
+          - Effect: Allow
+            Action:
+              - aps:RemoteWrite
+              - aps:GetSeries
+              - aps:GetLabels
+              - aps:GetMetricMetadata
+            Resource: !Sub "arn:${AWS::Partition}:aps:${AWS::Region}:${AWS::AccountId}:workspace/${PrometheusWorkspaceID}"
+          - Effect: Allow
+            Action:
+              - cloudwatch:PutMetricData
+            Resource: "*"
+            Condition:
+              StringEquals:
+                "cloudwatch:namespace": "testing.karpenter.sh/scale"
+  GithubActionsRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: GithubActionsRole
+      ManagedPolicyArns:
+        - !Ref GithubActionsPolicy
+        - !Ref GithubActionsPermissionsBoundary
+      MaxSessionDuration: 21600 # 6 hours is the max session for GHA
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Federated: !Ref GithubOIDCProvider
+            Action: "sts:AssumeRoleWithWebIdentity"
+            Condition:
+              StringEquals:
+                token.actions.githubusercontent.com:aud: sts.amazonaws.com
+              StringLike:
+                token.actions.githubusercontent.com:sub: !Sub "repo:${Repository}:ref:refs/heads/main"
```

Comment:
synced offline, need to validate after this PR is merged to check that this can work for snapshots before merge